### PR TITLE
handle the case where the block was created under us

### PR DESCRIPTION
**Pull request by abrookins**
Fixes a currently failing test: `TestThreadPoolTaskRunner.test_map_with_future_resolved_to_list`.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
 	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `mint.json` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `docs/mint.json` navigation.

### CODE REVIEW

File: src/prefect/results.py [hunk 1]
```diff
@@ -103,13 +103,17 @@ async def get_storage() -> WritableFileSystem:
...
-            return block
+            block = await block.save(name, overwrite=False)
         except ValueError as e:
             if "already in use" not in str(e):
                 raise e
+        except ObjectAlreadyExists:
+            # Another client created the block before we reached this line
+            pass
+        else:
+            block = await Block.load(default_storage_name)
```
Comment (cicdw) [inline src/prefect/results.py:114]: Doesn't this `else` only run if the `ObjectAlreadyExists` exception is _not_ raised?  Meaning you'd want this load to happen within the new except?

Comment (chrisguidry) [reply src/prefect/results.py:114]: +1, I think if we did hit `ObjectAlreadyExists` here, the `block` we return would be the unsaved `LocalFileSystem` block we created on L97, right?

Comment (abrookins) [reply src/prefect/results.py:114]: Ah yeah, I was tired 😂

Comment (abrookins) [reply src/prefect/results.py:114]: Nevertheless I continue to believe except/else wasn't a mistaken language feature

**Comment by abrookins** [comment]
closing this because we're fixing in #13611

**Pull request closed without merging**